components/table/interface.tsx

```diff
@@ -99,7 +99,7 @@ export interface TableProps<T> {
   expandRowByClick?: boolean;
   onExpandedRowsChange?: (expandedRowKeys: string[] | number[]) => void;
   onExpand?: (expanded: boolean, record: T) => void;
-  onChange?: (pagination: TablePaginationConfig | boolean, filters: string[], sorter: Object) => any;
+  onChange?: (pagination: TablePaginationConfig | boolean, filters: Record<keyof T, string[]>, sorter: Record<keyof T, any>) => any;
```

Comment:
Need to define a sort parameter type for third parameter

https://github.com/ant-design/ant-design/blob/8b6488256281a9db62378506e79b040a3d695bea/components/table/Table.tsx#L822-L828